evm/src/NttManager/NttManager.sol

```diff
@@ -291,18 +284,14 @@ contract NttManager is INttManager, NttManagerState {
                     // Since there is no standard way to query for burn fee amounts with burnable tokens,
                     // and NTT would be used on a per-token basis, implementing this functionality
                     // is left to integrating projects who may need to account for burn fees on their tokens.
-                    ERC20Burnable(token).burnFrom(msg.sender, amount);
-
-                    // query sender's token balance after transfer
-                    uint256 balanceAfter = _getTokenBalanceOf(token, msg.sender);
+                    ERC20Burnable(token).burn(amount);
 
-                    uint256 balanceDiff = balanceBefore - balanceAfter;
-                    if (balanceDiff != amount) {
-                        revert BurnAmountDifferentThanBalanceDiff(amount, balanceDiff);
+                    // tokens held by the contract after the operation should be the same as before
+                    uint256 balanceAfterBurn = _getTokenBalanceOf(token, address(this));
+                    if (balanceBefore != balanceAfterBurn) {
+                        revert BurnAmountDifferentThanBalanceDiff(balanceBefore, balanceAfterBurn);
                     }
                 }
-            } else {
-                revert InvalidMode(uint8(mode));
```

Comment:
Should we keep this check around in the transfer slow to prevent someone losing funds on a transfer that can't be redeemed on the destination chain?

Reply:
Actually it's an enum now so I don't think we an deploy with any other uint8 anyway